src/integrators/exponential_rk_integrators.jl

```diff
@@ -10,17 +10,17 @@
   integrator.k[2] = zero(integrator.fsalfirst)
 end
 
-@inline @muladd function perform_step!(integrator,cache::LawsonEulerConstantCache,f=integrator.f)
-  @unpack t,dt,uprev,u,k = integrator
+@inline function perform_step!(integrator,cache::LawsonEulerConstantCache,f=integrator.f)
+  @unpack t,dt,uprev,u = integrator
   rtmp = integrator.fsalfirst
   A = f[1]
-  u = expm(dt*A)*(@. uprev + dt*rtmp)
+  @muladd u = expm(dt*A)*(@. uprev + dt*rtmp)
```

Comment:
`@muladd` does not change or interfere with matrix multiplications. This generates the following expression:
```julia
julia> @macroexpand( @muladd u = expm(dt*A)*(@. uprev + dt*rtmp) )
:(u = expm(dt * A) * (muladd).(dt, rtmp, uprev))
```
So this should work as expected, shouldn't it? By the way, because of matrix multiplications I thought it would be better to only apply `muladd` as a dot call if both operators are dotted; i.e. expressions like `@muladd A*x .+ b` stay untouched whereas `@muladd a.*b.+c` leads to `(muladd).(a,b,c)`.